### ISSUE TITLE
Arch distro issue: Segmentation Fault on startup

### ISSUE THREAD

**Issue by oaken-source** [jliljebl/flowblade#445]
When I try to start flowblade-1.14 I get the following output:

```
 ~ $ flowblade 
FLOWBLADE MOVIE EDITOR 1.14
---------------------------
Launch script dir: /usr/bin
Running from installation...
modules path: /usr/share/pyshared/Flowblade
MLT found, version: 6.4.1
numpy version: 1.13.3
OS: Parabola GNU/Linux-libre
Python 2.7.14 (default, Sep 20 2017, 01:25:59) 
[GCC 7.2.0]
GTK+ version: 3.22.24
User dir: /home/andi/.flowblade/
Locale: en_US
Translations at /usr/share/locale were not found, using program root directory translations.
Use OS locale language.
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: flowblade.xml
Loading shortcuts: Flowblade
1600 900
Small height: True
Small width: False
Detecting environment...
---
video_codecs:
  - a64multi
  - a64multi5
  - alias_pix
  - amv
  - apng
  - asv1
  - asv2
  - avrp
  - avui
  - ayuv
  - bmp
  - cinepak
  - cljr
  - dnxhd
  - dpx
  - dvvideo
  - ffv1
  - ffvhuff
  - flashsv
  - flashsv2
  - flv
  - gif
  - h261
  - h263
  - h263p
  - huffyuv
  - jpeg2000
  - jpegls
  - ljpeg
  - mjpeg
  - mpeg1video
  - mpeg2video
  - mpeg4
  - msmpeg4v2
  - msmpeg4
  - msvideo1
  - pam
  - pbm
  - pcx
  - pgm
  - pgmyuv
  - png
  - ppm
  - prores
  - prores_aw
  - prores_ks
  - qtrle
  - r10k
  - r210
  - rawvideo
  - roqvideo
  - rv10
  - rv20
  - sgi
  - snow
  - sunrast
  - svq1
  - targa
  - tiff
  - utvideo
  - v210
  - v308
  - v408
  - v410
  - vc2
  - wrapped_avframe
  - wmv1
  - wmv2
  - xbm
  - xface
  - xwd
  - y41p
  - yuv4
  - zlib
  - zmbv
  - libopenjpeg
  - libschroedinger
  - libtheora
  - libvpx
  - libvpx-vp9
  - libwebp_anim
  - libwebp
  - libx264
  - libx264rgb
  - libx265
  - libxvid
  - h264_nvenc
  - h264_vaapi
  - nvenc
  - nvenc_h264
  - nvenc_hevc
  - hevc_nvenc
  - hevc_vaapi
  - mjpeg_vaapi
  - mpeg2_vaapi
  - vp8_vaapi
...
---
audio_codecs:
  - comfortnoise
  - s302m
  - aac
  - ac3
  - ac3_fixed
  - alac
  - dca
  - eac3
  - flac
  - g723_1
  - mlp
  - mp2
  - mp2fixed
  - nellymoser
  - opus
  - real_144
  - sonic
  - sonicls
  - truehd
  - tta
  - vorbis
  - wavpack
  - wmav1
  - wmav2
  - pcm_alaw
  - pcm_f32be
  - pcm_f32le
  - pcm_f64be
  - pcm_f64le
  - pcm_mulaw
  - pcm_s8
  - pcm_s8_planar
  - pcm_s16be
  - pcm_s16be_planar
  - pcm_s16le
  - pcm_s16le_planar
  - pcm_s24be
  - pcm_s24daud
  - pcm_s24le
  - pcm_s24le_planar
  - pcm_s32be
  - pcm_s32le
  - pcm_s32le_planar
  - pcm_s64be
  - pcm_s64le
  - pcm_u8
  - pcm_u16be
  - pcm_u16le
  - pcm_u24be
  - pcm_u24le
  - pcm_u32be
  - pcm_u32le
  - roq_dpcm
  - adpcm_adx
  - g722
  - g726
  - adpcm_ima_qt
  - adpcm_ima_wav
  - adpcm_ms
  - adpcm_swf
  - adpcm_yamaha
  - libgsm
  - libgsm_ms
  - libmp3lame
  - libopencore_amrnb
  - libopus
  - libspeex
  - libvorbis
...
---
formats:
  - a64
  - ac3
  - adts
  - adx
  - aiff
  - amr
  - apng
  - asf
  - ass
  - ast
  - asf_stream
  - au
  - avi
  - avm2
  - bit
  - caf
  - cavsvideo
  - crc
  - dash
  - data
  - daud
  - dirac
  - dnxhd
  - dts
  - dv
  - eac3
  - f4v
  - ffm
  - ffmetadata
  - fifo
  - filmstrip
  - flac
  - flv
  - framecrc
  - framehash
  - framemd5
  - g722
  - g723_1
  - gif
  - gsm
  - gxf
  - h261
  - h263
  - h264
  - hash
  - hds
  - hevc
  - hls
  - ico
  - ilbc
  - image2
  - image2pipe
  - ipod
  - ircam
  - ismv
  - ivf
  - jacosub
  - latm
  - lrc
  - m4v
  - md5
  - matroska
  - matroska
  - microdvd
  - mjpeg
  - mlp
  - mmf
  - mov
  - mp2
  - mp3
  - mp4
  - mpeg
  - vcd
  - mpeg1video
  - dvd
  - svcd
  - mpeg2video
  - vob
  - mpegts
  - mpjpeg
  - mxf
  - mxf_d10
  - mxf_opatom
  - null
  - nut
  - oga
  - ogg
  - ogv
  - oma
  - opus
  - alaw
  - mulaw
  - f64be
  - f64le
  - f32be
  - f32le
  - s32be
  - s32le
  - s24be
  - s24le
  - s16be
  - s16le
  - s8
  - u32be
  - u32le
  - u24be
  - u24le
  - u16be
  - u16le
  - u8
  - psp
  - rawvideo
  - rm
  - roq
  - rso
  - rtp
  - rtp_mpegts
  - rtsp
  - sap
  - scc
  - segment
  - stream_segment,ssegment
  - singlejpeg
  - smjpeg
  - smoothstreaming
  - sox
  - spx
  - spdif
  - srt
  - swf
  - tee
  - 3g2
  - 3gp
  - mkvtimestamp_v2
  - truehd
  - tta
  - uncodedframecrc
  - vc1
  - vc1test
  - voc
  - w64
  - wav
  - webm
  - webm_dash_manifest
  - webm_chunk
  - webp
  - webvtt
  - wtv
  - wv
  - yuv4mpegpipe
  - alsa
  - fbdev
  - oss
  - pulse
  - sdl,sdl2
  - v4l2
...
MLT detection succeeded, 156 formats, 96 video codecs and 68 audio codecs found.
522 MLT services found.
Loading render profiles...
Loading filters...
Brightness dropped, MLT version too low for this filter.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
G'MIC found
Natron not found
Player initialized with profile:  DV/DVD PAL
Segmentation fault (core dumped)
```

I am using the AUR package.

I tried debugging the problem with gdb, and got this backtrace:

```
Thread 1 "python2" received signal SIGSEGV, Segmentation fault.
0x00007ffff7bc39b0 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
(gdb) back
#0  0x00007ffff7bc39b0 in pthread_mutex_lock () at /usr/lib/libpthread.so.0
#1  0x00007fffef98771a in XrmQGetResource () at /usr/lib/libX11.so.6
#2  0x00007fffef963ba6 in XGetDefault () at /usr/lib/libX11.so.6
#3  0x00007ffff0ae5808 in _XcursorGetDisplayInfo () at /usr/lib/libXcursor.so.1
#4  0x00007ffff0ae5849 in XcursorSupportsARGB () at /usr/lib/libXcursor.so.1
#5  0x00007ffff0ae81e1 in XcursorNoticeCreateBitmap () at /usr/lib/libXcursor.so.1
#6  0x00007fffef95ea61 in XCreatePixmap () at /usr/lib/libX11.so.6
#7  0x00007ffff1c03b76 in  () at /usr/lib/libgdk-3.so.0
#8  0x00007ffff1c04111 in  () at /usr/lib/libgdk-3.so.0
#9  0x00007fffeca12462 in  () at /usr/lib/libgtk-3.so.0
#10 0x00007fffeca19c9d in  () at /usr/lib/libgtk-3.so.0
#11 0x00007ffff532c6f5 in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#12 0x00007ffff5340188 in  () at /usr/lib/libgobject-2.0.so.0
#13 0x00007ffff5344696 in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#14 0x00007ffff5345920 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#15 0x00007fffeca09f3c in gtk_widget_realize () at /usr/lib/libgtk-3.so.0
#16 0x00007fffeca180d1 in  () at /usr/lib/libgtk-3.so.0
#17 0x00007ffff532c6f5 in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#18 0x00007ffff5340188 in  () at /usr/lib/libgobject-2.0.so.0
#19 0x00007ffff5344696 in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#20 0x00007ffff5345920 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#21 0x00007fffeca03c3b in gtk_widget_show () at /usr/lib/libgtk-3.so.0
#22 0x00007ffff4dee1c8 in ffi_call_unix64 () at /usr/lib/libffi.so.6
#23 0x00007ffff4dedc2a in ffi_call () at /usr/lib/libffi.so.6
#24 0x00007ffff57c5a44 in  () at /usr/lib/python2.7/site-packages/gi/_gi.so
#25 0x00007ffff57c76c8 in  () at /usr/lib/python2.7/site-packages/gi/_gi.so
#26 0x00007ffff57bb4fa in  () at /usr/lib/python2.7/site-packages/gi/_gi.so
#27 0x00007ffff74cdd93 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0
#28 0x00007ffff74aac6e in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#29 0x00007ffff7504886 in PyEval_EvalCodeEx () at /usr/lib/libpython2.7.so.1.0
#30 0x00007ffff747d80f in function_call.lto_priv () at /usr/lib/libpython2.7.so.1.0
#31 0x00007ffff74cdd93 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0
#32 0x00007ffff746244f in instancemethod_call.lto_priv () at /usr/lib/libpython2.7.so.1.0
#33 0x00007ffff74cdd93 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0
#34 0x00007ffff7503f61 in PyEval_CallObjectWithKeywords () at /usr/lib/libpython2.7.so.1.0
#35 0x00007ffff746627f in PyInstance_New () at /usr/lib/libpython2.7.so.1.0
#36 0x00007ffff74cdd93 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0
#37 0x00007ffff74aac6e in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#38 0x00007ffff74a5b50 in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#39 0x00007ffff74a5b50 in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#40 0x00007ffff7504886 in PyEval_EvalCodeEx () at /usr/lib/libpython2.7.so.1.0
#41 0x00007ffff751a05a in PyEval_EvalCode () at /usr/lib/libpython2.7.so.1.0
#42 0x00007ffff75217f1 in run_mod () at /usr/lib/libpython2.7.so.1.0
#43 0x00007ffff75230d5 in PyRun_FileExFlags () at /usr/lib/libpython2.7.so.1.0
#44 0x00007ffff75232aa in PyRun_SimpleFileExFlags () at /usr/lib/libpython2.7.so.1.0
#45 0x00007ffff7511863 in Py_Main () at /usr/lib/libpython2.7.so.1.0
#46 0x00007ffff7823f6a in __libc_start_main () at /usr/lib/libc.so.6
#47 0x000055555555478a in _start ()
```

I'm a bit at a loss how to locate the issue. Any help is appreciated :)

**Comment by jliljebl** [comment]
Are you using MLT build with SDL2 instead of SDL 1.2. SDL2 breaks Flowblade currently with the problems appearing in X11 like in your backtrace.

**Comment by oaken-source** [comment]
I don't think so, but I am not sure. I am using the original arch mlt package at the moment.
How do I check? `ldd /usr/lib/libmlt.so` does not include any reference to SDL at all.

**Comment by dvdlvr** [comment]
You may try doing (Adapted for your own path, of course)
ldd /usr/lib/x86_64-linux-gnu/mlt/libmltsdl.so | grep libSDL

I get this:
        libSDL-1.2.so.0 => /usr/lib/x86_64-linux-gnu/libSDL-1.2.so.0 (0x00007f93e84dd000)

**Comment by oaken-source** [comment]
This is what I get, too:
```
~ $ ldd /usr/lib/mlt/libmltsdl.so | grep SDL
	libSDL-1.2.so.0 => /usr/lib/libSDL-1.2.so.0 (0x00007f4377872000)
	libSDL_image-1.2.so.0 => /usr/lib/libSDL_image-1.2.so.0 (0x00007f4377654000)
```

**Comment by jliljebl** [comment]
Ok, so problem is not with SDL version then. Flowblade works with GTK 3.22 on my system, so GTK version should not be an issue either. We will just need to leave this open for now and wait for some additional information.

**Comment by oaken-source** [comment]
If there is something I can do to help debug this, then let me know :)
thanks.

**Comment by jliljebl** [comment]
Stack trace says that is this somehow related to GtK->GDK->X11 but I currently have no idea what is going on exactly, GTK should work. If I get some idea on where to look exactly, I'll post here and we can see if we could test something that provides additional info on the subject.

**Comment by mindfulmonk** [comment]
I'm encountering the same issue on arch. 

```
MLT detection succeeded, 160 formats, 97 video codecs and 69 audio codecs found.
526 MLT services found.
Loading render profiles...
Loading filters...
Brightness dropped, MLT version too low for this filter.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
G'MIC found
Natron not found
Player initialized with profile:  DV/DVD PAL
Selected color NOT detected
BG color detected
[swscaler @ 0x7fd7f8240e60] Warning: data is not aligned! This can lead to a speed loss
[1]    4036 segmentation fault (core dumped)  flowblade
```

Installed through https://aur.archlinux.org/packages/flowblade/

**Comment by oberon2007** [comment]
Any progress on this? **flowblade** is still unusable on Arch and Manjaro.

**Comment by jliljebl** [comment]
I need a gdb stack trace to even start guessing what the problem is. Here are some instructions: https://github.com/jliljebl/flowblade/blob/master/flowblade-trunk/docs/GDB.md

**Comment by oaken-source** [comment]
A stacktrace is included in the original issue description :)

**Comment by jliljebl** [comment]
Yeah true, I remember this now. No, I'm not seeing this and I do not know how to replicate the issue.  I'm guessing KDEnlive is working because this a GTK Issue, not an MLT issue.

This can only be effectively debugged from affected systems and I do not have resources to set up Arch and start chasing this, that could easily kill my whole devime budget for the cycle.

This is 100% guaranteed to be an issue that cannot be fixed with Flowblade source code changes. 

This is also where the community aspect of Linux desktop will either help us or fail us. We need someone on Arch system to be willing and capable to chase this down.  I cannot reasonably know why realizing some GTK widget on Arch causes crash in X11. I repeat, this is not bug that can fixed by changing Flowblade Python source code.

**Comment by rredesigns** [comment]
Ok, so I'm jumping in here. This is what I get with gdb (just the relevant `part):

```Reading symbols from python2...(no debugging symbols found)...done.
(gdb) run /mnt/Storage/Linux_files/flowblade/flowblade 
Starting program: /usr/bin/python2 /mnt/Storage/Linux_files/flowblade/flowblade
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
FLOWBLADE MOVIE EDITOR 1.14
---------------------------
Launch script dir: /mnt/Storage/Linux_files/flowblade
Running from filesystem...
MLT found, version: 6.4.1
[New Thread 0x7fffe94f5700 (LWP 8683)]
[New Thread 0x7fffe8cf4700 (LWP 8684)]
numpy version: 1.13.3
OS: Manjaro Linux
Python 2.7.14 (default, Sep 20 2017, 01:25:59) 
[GCC 7.2.0]
GTK+ version: 3.22.26
User dir: /home/rre/.flowblade/
Locale: es_AR
Found translations at /mnt/Storage/Linux_files/flowblade/Flowblade/locale/, using those.
Use OS locale language.
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: Flowblade Default
1366 768
Small height: True
Small width: True
[New Thread 0x7fffb60a5700 (LWP 8685)]
Detecting environment...

...
... Codecs and services ommitted here.
...

Brightness dropped, MLT version too low for this filter.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
[New Thread 0x7fffb03bd700 (LWP 8686)]
[New Thread 0x7fffaf095700 (LWP 8687)]
G'MIC found
Natron not found
Player initialized with profile:  DV/DVD PAL
[Thread 0x7fffaf095700 (LWP 8687) exited]
[New Thread 0x7fffaf095700 (LWP 8689)]
[New Thread 0x7fffae5be700 (LWP 8690)]
[New Thread 0x7fffaddbd700 (LWP 8691)]
[New Thread 0x7fffad5bc700 (LWP 8692)]

(flowblade:8679): Gtk-WARNING **: Theme directory  48x48/devices of theme Super-Ultra-Flat-Numix-Remix-Majestic has no size field


(flowblade:8679): Gtk-WARNING **: Theme directory 64@2x/panel of theme Super-Ultra-Flat-Numix-Remix has no size field

[New Thread 0x7fffacc1a700 (LWP 8693)]
[New Thread 0x7fff97fff700 (LWP 8694)]
[New Thread 0x7fff977fe700 (LWP 8695)]
[New Thread 0x7fff96b8b700 (LWP 8696)]
[New Thread 0x7fff9638a700 (LWP 8697)]
Selected color NOT detected
BG color detected
[New Thread 0x7fff95adc700 (LWP 8698)]
[Thread 0x7fff95adc700 (LWP 8698) exited]
[New Thread 0x7fff95adc700 (LWP 8701)]
[New Thread 0x7fff9512b700 (LWP 8702)]
[swscaler @ 0x7fff7829e900] Warning: data is not aligned! This can lead to a speed loss
[New Thread 0x7fff67fff700 (LWP 8704)]
[Thread 0x7fff95adc700 (LWP 8701) exited]
[Thread 0x7fff67fff700 (LWP 8704) exited]
[Thread 0x7fff9512b700 (LWP 8702) exited]
[New Thread 0x7fff9512b700 (LWP 8705)]
[Thread 0x7fff9512b700 (LWP 8705) exited]
[New Thread 0x7fff9512b700 (LWP 8706)]
[New Thread 0x7fff67fff700 (LWP 8707)]
[New Thread 0x7fff95adc700 (LWP 8709)]

Thread 1 "python2" received signal SIGSEGV, Segmentation fault.
0x00007ffff7bc39b0 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
(gdb) 
```

**Comment by jliljebl** [comment]
OK, so it is clear that crash happens when methot show() is called on a GTK
widget.

Now we need to find out where. We need to start putting print staments in
code to find the exact show() that causes the crash.

First we need to add staments like print "1",  print "2" on file app.py
method main () to find out how far down we get in that method and then
follow the crashing statement until we find the exact line in Flowblade
code that causes the crash.



3.12.2017 9.36 ip. "rredesigns" <notifications@github.com> kirjoitti:

Ok, so I'm jumping in here. This is what I get with gdb (just the relevant
`part):

(gdb) run /mnt/Storage/Linux_files/flowblade/flowblade
Starting program: /usr/bin/python2 /mnt/Storage/Linux_files/flowblade/flowblade
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
FLOWBLADE MOVIE EDITOR 1.14
---------------------------
Launch script dir: /mnt/Storage/Linux_files/flowblade
Running from filesystem...
MLT found, version: 6.4.1
[New Thread 0x7fffe94f5700 (LWP 8683)]
[New Thread 0x7fffe8cf4700 (LWP 8684)]
numpy version: 1.13.3
OS: Manjaro Linux
Python 2.7.14 (default, Sep 20 2017, 01:25:59)
[GCC 7.2.0]
GTK+ version: 3.22.26
User dir: /home/rre/.flowblade/
Locale: es_AR
Found translations at
/mnt/Storage/Linux_files/flowblade/Flowblade/locale/, using those.
Use OS locale language.
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: Flowblade Default
1366 768
Small height: True
Small width: True
[New Thread 0x7fffb60a5700 (LWP 8685)]
Detecting environment...

...
... Codecs and services ommitted here.
...

Brightness dropped, MLT version too low for this filter.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
[New Thread 0x7fffb03bd700 (LWP 8686)]
[New Thread 0x7fffaf095700 (LWP 8687)]
G'MIC found
Natron not found
Player initialized with profile:  DV/DVD PAL
[Thread 0x7fffaf095700 (LWP 8687) exited]
[New Thread 0x7fffaf095700 (LWP 8689)]
[New Thread 0x7fffae5be700 (LWP 8690)]
[New Thread 0x7fffaddbd700 (LWP 8691)]
[New Thread 0x7fffad5bc700 (LWP 8692)]

(flowblade:8679): Gtk-WARNING **: Theme directory  48x48/devices of
theme Super-Ultra-Flat-Numix-Remix-Majestic has no size field


(flowblade:8679): Gtk-WARNING **: Theme directory 64@2x/panel of theme
Super-Ultra-Flat-Numix-Remix has no size field

[New Thread 0x7fffacc1a700 (LWP 8693)]
[New Thread 0x7fff97fff700 (LWP 8694)]
[New Thread 0x7fff977fe700 (LWP 8695)]
[New Thread 0x7fff96b8b700 (LWP 8696)]
[New Thread 0x7fff9638a700 (LWP 8697)]
Selected color NOT detected
BG color detected
[New Thread 0x7fff95adc700 (LWP 8698)]
[Thread 0x7fff95adc700 (LWP 8698) exited]
[New Thread 0x7fff95adc700 (LWP 8701)]
[New Thread 0x7fff9512b700 (LWP 8702)]
[swscaler @ 0x7fff7829e900] Warning: data is not aligned! This can
lead to a speed loss
[New Thread 0x7fff67fff700 (LWP 8704)]
[Thread 0x7fff95adc700 (LWP 8701) exited]
[Thread 0x7fff67fff700 (LWP 8704) exited]
[Thread 0x7fff9512b700 (LWP 8702) exited]
[New Thread 0x7fff9512b700 (LWP 8705)]
[Thread 0x7fff9512b700 (LWP 8705) exited]
[New Thread 0x7fff9512b700 (LWP 8706)]
[New Thread 0x7fff67fff700 (LWP 8707)]
[New Thread 0x7fff95adc700 (LWP 8709)]

Thread 1 "python2" received signal SIGSEGV, Segmentation fault.
0x00007ffff7bc39b0 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
(gdb)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/jliljebl/flowblade/issues/445#issuecomment-348808128>,
or mute the thread

**Comment by rredesigns** [comment]
No problem. I can do that. Just let me know, so we don't do the same task twice. xD

**Comment by jliljebl** [comment]
On my system the crash does not happen, I need you to show the crashing
line of code.

4.12.2017 3.03 ip. "rredesigns" <notifications@github.com> kirjoitti:

No problem. I can do that. Just let me know, so we don't do the same task
twice. xD

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/jliljebl/flowblade/issues/445#issuecomment-348956187>,
or mute the thread

**Comment by rredesigns** [comment]
Ok, I have my last exam tomorrow. After that I'll get to it.

**Comment by jliljebl** [comment]
I'm merging this with #466 , it is very likely the same Issue.

**Comment by jliljebl** [comment]
Ok, this NOT the same issue, reopening, debugging help still needed.

**Comment by rredesigns** [comment]
I will take a look at the scripts in these days. :)

**Comment by rredesigns** [comment]
Ok, so it gets down to here. It prints "12" then crashes.![here.](https://user-images.githubusercontent.com/10718720/34308907-30e1b7b4-e72e-11e7-8be6-c7fd16d2f4e9.png)

Here's the terminal output 
![image](https://user-images.githubusercontent.com/10718720/34308957-646242ac-e72e-11e7-9c18-d9e4b4e31709.png)


I don't know how to inspect the Gtk stuff. Maybe downgrading the toolkit would do? I will try that later on.

**Comment by kppmsd** [comment]
I did some digging on this, and found out that if I comment out `launch_player()` from main() in app.py, the application doesn't segfault, the GUI finishes startup, and things seem to work (apart from the player obviously).

Am I wrong on thinking this might therefore have something to do with SDL on Arch?

I'll try to dig deeper on this when I have time. Any ideas for further investigation are appreciated.

**Comment by rredesigns** [comment]
I downgraded SDL 3 versions, and it still segfaults. :|

I guess the best way to figure this out is that the main dev or some collaborator try it on a virtual machine with Arch (I would suggest Manjaro though, for easier/faster setup).

**Comment by kppmsd** [comment]
It seems I solved the issue for myself.

It turned out, that at least on my case MLT was built against SDL2, as jliljebl already suggested on second post. After building MLT against SDL1, flowblade starts fine, and everything including video playback seems to work.

Could someone else on Arch try if this solution works for you? Unfortunately I couldn't find a simple way to force building MLT against SDL1, so I had to forcibly uninstall SDL2 from my system before rebuilding MLT. Building the AUR package in clean chroot should also work, since at least mlt-git only depends on SDL1 packages. Also make sure you have sdl_image installed, as it is required for MLT's SDL backend.

I still encountered few crashes at startup, and there's segfault after exiting the application most of the time, but I think these are unrelated. I'm using latest flowblade from git.

**Comment by oaken-source** [comment]
Hi, I took another look to see which sdl versions are used by our distributions (parabola GNU/linux-libre) mlt packages:

```
$> for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -qi sdl2 && echo $lib; done
/usr/lib/mlt/libmltavformat.so
```

the complete list of sdl refences is:
```
$ for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -i sdl && echo $lib; done
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fa9738a6000)
/usr/lib/mlt/libmltavformat.so
	libSDL-1.2.so.0 => /usr/lib/libSDL-1.2.so.0 (0x00007f90b5802000)
	libSDL_image-1.2.so.0 => /usr/lib/libSDL_image-1.2.so.0 (0x00007f90b55e4000)
/usr/lib/mlt/libmltsdl.so
```

I manually confirmed via lddtree, that mlt itself is indeed built against SDL1. the only thing pulling SDL2 into the mix and into libmltavformat.so is libavdevice.so, which comes from ffmpeg.

@kppmsd could you run the commands quoted above on your system to check if the version of ffmpeg you use in mlt is built against sdl2? because if so, I could remove this from the list of possible causes :)

**Comment by kppmsd** [comment]
Yes, it's the same for me:
```
[~] $ for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -i sdl && echo $lib; done
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fa43b8a0000)
/usr/lib/mlt/libmltavformat.so
	libSDL-1.2.so.0 => /usr/lib/libSDL-1.2.so.0 (0x00007f883a2ad000)
	libSDL_image-1.2.so.0 => /usr/lib/libSDL_image-1.2.so.0 (0x00007f883a08f000)
/usr/lib/mlt/libmltsdl.so
```

I also checked that flowblade version 1.14 works for me as well, not just the latest git..

**Comment by jliljebl** [comment]
Ok, I'll keep this open until the new video consumer is available.

**Comment by popsUlfr** [comment]
The following patches should force SDL1 to be used. I get the same ldd output but flowblade still segfaults for me :(

`sdl2begone.patch`
```diff
diff --git a/src/melt/Makefile b/src/melt/Makefile
index f58f852f..b6546b46 100644
--- a/src/melt/Makefile
+++ b/src/melt/Makefile
@@ -10,7 +10,7 @@ LDFLAGS += -L../framework -lmlt -lpthread
 SRCS := $(OBJS:.o=.c)
 
 ifeq (, $(findstring MELT_NOSDL, $(CFLAGS)))
-ifeq (yes, $(shell pkg-config --exists sdl2 && echo yes))
+ifeq (yes, $(shell pkg-config --exists sdl2 && echo no))
 CFLAGS += -DHAVE_SDL2
 CFLAGS  += $(shell pkg-config --cflags sdl2)
 LDFLAGS += $(shell pkg-config --libs sdl2)
diff --git a/src/modules/sdl/configure b/src/modules/sdl/configure
index 0f422d56..bf8a4ec8 100755
--- a/src/modules/sdl/configure
+++ b/src/modules/sdl/configure
@@ -2,7 +2,7 @@
 
 if [ "$help" != "1" ]; then
 
-	if pkg-config --exists sdl2 > /dev/null 2>&1 ; then
+	if false && pkg-config --exists sdl2 > /dev/null 2>&1 ; then
 		echo " - using SDL2"
 		echo "USE_PKG_CONFIG=1" > config.mak
 		echo "HAVE_SDL2=1" >> config.mak
```

`mlt-git-sdl2begone-PKGBUILD.patch` for the PKGBUILD file
```diff
diff --git a/PKGBUILD b/PKGBUILD
index 9e66d67..634839a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,8 +1,8 @@
 pkgname=('mlt-git')
 _srcname='mlt'
 pkgdesc='Multimedia Framework'
-pkgver='r1'
-pkgrel='1'
+pkgver=r4567.1515478749.d1fabee5
+pkgrel=1
 arch=('i686' 'x86_64')
 url="https://github.com/mltframework/${_srcname}"
 license=('GPL2')
@@ -40,8 +40,9 @@ makedepends=(
 provides=("${pkgname[0]%-git}")
 conflicts=("${pkgname[0]%-git}")
 
-source=("${_srcname}::git+${url}.git")
-sha512sums=('SKIP')
+source=("${_srcname}::git+${url}.git"
+'sdl2begone.patch')
+sha512sums=('SKIP' 'SKIP')
 
 pkgver() {
     cd "${srcdir}/${_srcname}"
@@ -52,6 +53,11 @@ pkgver() {
         "$( git rev-parse --short 'HEAD' )"
 }
 
+prepare() {
+	cd "${srcdir}/${_srcname}"
+	patch -Np1 -i "${srcdir}/sdl2begone.patch"
+}
+
 build() {
     cd "${srcdir}/${_srcname}"
 ```

**Comment by deimidis** [comment]
I had this result:

$ for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -i sdl && echo $lib; done
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fdb87bbb000)
/usr/lib/mlt/libmltavformat.so
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007f97a410e000)
/usr/lib/mlt/libmltsdl.so
[deimidis@binaural ~]$ for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -i sdl && echo $lib; done
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007f7180aea000)
/usr/lib/mlt/libmltavformat.so
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007f118b5b8000)
/usr/lib/mlt/libmltsdl.so

Using Manjaro-testing version

**Comment by jliljebl** [comment]
Does updating to MLT 6.6.0 fix the issue? See here:

https://github.com/jliljebl/flowblade/blob/master/flowblade-trunk/docs/SDL_2_AND_MLT_6_6_0.md

**Comment by popsUlfr** [comment]
@jliljebl 

mlt 6.6.0 has been packaged in extra on arch and that one works indeed! Flowblade starts and works just like in the past (from what I could test). mlt-git makes flowblade segfault, replacing `mlt-git` with `mlt` in the `PKGBUILD` works though.

**Comment by jliljebl** [comment]
Great, I'll keep this open for info for a bit more.

**Comment by oberon2007** [comment]
Very good! I have adjusted the dependency in the AUR PKGBUID, just we don't have a way of forcing it for people who already have mlt-git installed, since the package 'provides' mlt.
Anyway confirming that it works fine now. Thank you!

**Comment by mrkaato** [comment]
Thanks, yes, it works now. I removed MLT-git and Flowblade and I installed MLT and re-installed Flowblade 1.14-1
No instructions for coming to the website?

**Comment by jliljebl** [comment]
We can close this now, Arch working.

**Comment by dcanelhas** [comment]
Is there some work-around needed to get this working on Arch? 

I removed mlt, flowblade, mlt-pythonbindings, purged the cache and reinstalled flowblade from aur, letting it pull in the dependencies as needed using the PKGBUILD from the AUR. Currently i get a segfault at start, running:

```bash
$ gdb --args python2 /usr/bin/flowblade
GNU gdb (GDB) 8.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python2...(no debugging symbols found)...done.
(gdb) r
Starting program: /usr/bin/python2 /usr/bin/flowblade
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
FLOWBLADE MOVIE EDITOR 1.16
---------------------------
modules path: /usr/share/pyshared/Flowblade
MLT found, version: 6.8.0
[New Thread 0x7fffe5831700 (LWP 7581)]
[New Thread 0x7fffe5030700 (LWP 7582)]
numpy version: 1.14.3
OS: Arch Linux
Python 2.7.15 (default, May  1 2018, 20:16:04) 
[GCC 7.3.1 20180406]
GTK+ version: 3.22.30
User dir: /home/danielcanelhas/.flowblade/
Locale: en_US
Translations at /usr/share/locale were not found, using program root directory translations.
Use OS locale language.
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: flowblade.xml
Loading shortcuts: Flowblade
3840 1080
Small height: False
Small width: False
[New Thread 0x7fffbb6d1700 (LWP 7583)]
Detecting environment...
---
video_codecs:
  - a64multi
  - a64multi5
  - alias_pix
  - amv
  - apng
  - asv1
  - asv2
  - avrp
  - avui
  - ayuv
  - bmp
  - cinepak
  - cljr
  - dnxhd
  - dpx
  - dvvideo
  - ffv1
  - ffvhuff
  - fits
  - flashsv
  - flashsv2
  - flv
  - gif
  - h261
  - h263
  - h263p
  - huffyuv
  - jpeg2000
  - jpegls
  - ljpeg
  - magicyuv
  - mjpeg
  - mpeg1video
  - mpeg2video
  - mpeg4
  - msmpeg4v2
  - msmpeg4
  - msvideo1
  - pam
  - pbm
  - pcx
  - pgm
  - pgmyuv
  - png
  - ppm
  - prores
  - prores_aw
  - prores_ks
  - qtrle
  - r10k
  - r210
  - rawvideo
  - roqvideo
  - rv10
  - rv20
  - sgi
  - snow
  - sunrast
  - svq1
  - targa
  - tiff
  - utvideo
  - v210
  - v308
  - v408
  - v410
  - vc2
  - wrapped_avframe
  - wmv1
  - wmv2
  - xbm
  - xface
  - xwd
  - y41p
  - yuv4
  - zlib
  - zmbv
  - libopenjpeg
  - libtheora
  - libvpx
  - libvpx-vp9
  - libwebp_anim
  - libwebp
  - libx264
  - libx264rgb
  - libx265
  - libxvid
  - h264_nvenc
  - h264_omx
  - h264_vaapi
  - nvenc
  - nvenc_h264
  - nvenc_hevc
  - hevc_nvenc
  - hevc_vaapi
  - mjpeg_vaapi
  - mpeg2_vaapi
  - vp8_vaapi
  - vp9_vaapi
...
---
audio_codecs:
  - comfortnoise
  - s302m
  - aac
  - ac3
  - ac3_fixed
  - alac
  - aptx
  - aptx_hd
  - dca
  - eac3
  - flac
  - g723_1
  - mlp
  - mp2
  - mp2fixed
  - nellymoser
  - opus
  - real_144
  - sbc
  - sonic
  - sonicls
  - truehd
  - tta
  - vorbis
  - wavpack
  - wmav1
  - wmav2
  - pcm_alaw
  - pcm_f32be
  - pcm_f32le
  - pcm_f64be
  - pcm_f64le
  - pcm_mulaw
  - pcm_s8
  - pcm_s8_planar
  - pcm_s16be
  - pcm_s16be_planar
  - pcm_s16le
  - pcm_s16le_planar
  - pcm_s24be
  - pcm_s24daud
  - pcm_s24le
  - pcm_s24le_planar
  - pcm_s32be
  - pcm_s32le
  - pcm_s32le_planar
  - pcm_s64be
  - pcm_s64le
  - pcm_u8
  - pcm_u16be
  - pcm_u16le
  - pcm_u24be
  - pcm_u24le
  - pcm_u32be
  - pcm_u32le
  - roq_dpcm
  - adpcm_adx
  - g722
  - g726
  - g726le
  - adpcm_ima_qt
  - adpcm_ima_wav
  - adpcm_ms
  - adpcm_swf
  - adpcm_yamaha
  - libgsm
  - libgsm_ms
  - libmp3lame
  - libopencore_amrnb
  - libopus
  - libspeex
  - libvorbis
...
---
formats:
  - a64
  - ac3
  - adts
  - adx
  - aiff
  - amr
  - apng
  - aptx
  - aptx_hd
  - asf
  - ass
  - ast
  - asf_stream
  - au
  - avi
  - avm2
  - bit
  - caf
  - cavsvideo
  - codec2
  - codec2raw
  - crc
  - dash
  - data
  - daud
  - dirac
  - dnxhd
  - dts
  - dv
  - eac3
  - f4v
  - ffmetadata
  - fifo
  - fifo_test
  - filmstrip
  - fits
  - flac
  - flv
  - framecrc
  - framehash
  - framemd5
  - g722
  - g723_1
  - g726
  - g726le
  - gif
  - gsm
  - gxf
  - h261
  - h263
  - h264
  - hash
  - hds
  - hevc
  - hls
  - ico
  - ilbc
  - image2
  - image2pipe
  - ipod
  - ircam
  - ismv
  - ivf
  - jacosub
  - latm
  - lrc
  - m4v
  - md5
  - matroska
  - matroska
  - microdvd
  - mjpeg
  - mlp
  - mmf
  - mov
  - mp2
  - mp3
  - mp4
  - mpeg
  - vcd
  - mpeg1video
  - dvd
  - svcd
  - mpeg2video
  - vob
  - mpegts
  - mpjpeg
  - mxf
  - mxf_d10
  - mxf_opatom
  - null
  - nut
  - oga
  - ogg
  - ogv
  - oma
  - opus
  - alaw
  - mulaw
  - f64be
  - f64le
  - f32be
  - f32le
  - s32be
  - s32le
  - s24be
  - s24le
  - s16be
  - s16le
  - s8
  - u32be
  - u32le
  - u24be
  - u24le
  - u16be
  - u16le
  - u8
  - psp
  - rawvideo
  - rm
  - roq
  - rso
  - rtp
  - rtp_mpegts
  - rtsp
  - sap
  - sbc
  - scc
  - film_cpk
  - segment
  - stream_segment,ssegment
  - singlejpeg
  - smjpeg
  - smoothstreaming
  - sox
  - spx
  - spdif
  - srt
  - sup
  - swf
  - tee
  - 3g2
  - 3gp
  - mkvtimestamp_v2
  - truehd
  - tta
  - uncodedframecrc
  - vc1
  - vc1test
  - voc
  - w64
  - wav
  - webm
  - webm_dash_manifest
  - webm_chunk
  - webp
  - webvtt
  - wtv
  - wv
  - yuv4mpegpipe
  - alsa
  - fbdev
  - oss
  - pulse
  - sdl,sdl2
  - v4l2
  - xv
...
MLT detection succeeded, 167 formats, 99 video codecs and 72 audio codecs found.
556 MLT services found.
Loading render profiles...
Loading filters...
Brightness dropped, MLT version too low for this filter.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
[New Thread 0x7fffac9e4700 (LWP 7589)]
[New Thread 0x7fffa775e700 (LWP 7590)]
[New Thread 0x7fffa6773700 (LWP 7592)]
G'MIC found
Natron not found
Player initialized with profile:  HD 1080p 30 fps
[Thread 0x7fffa6773700 (LWP 7592) exited]
[New Thread 0x7fffa6773700 (LWP 7594)]
[New Thread 0x7fffa5f72700 (LWP 7595)]
[New Thread 0x7fffa5771700 (LWP 7596)]
[New Thread 0x7fffa4f70700 (LWP 7597)]
[New Thread 0x7fff8ffff700 (LWP 7598)]
[New Thread 0x7fff8f7fe700 (LWP 7599)]
[New Thread 0x7fff8effd700 (LWP 7600)]
Selected color NOT detected
BG color detected
Create SDL1 consumer...
[New Thread 0x7fff8e7fc700 (LWP 7601)]
[Thread 0x7fff8e7fc700 (LWP 7601) exited]
[New Thread 0x7fff8e7fc700 (LWP 7602)]
[New Thread 0x7fff8d812700 (LWP 7603)]
[New Thread 0x7fff8cdff700 (LWP 7605)]
[Thread 0x7fff8e7fc700 (LWP 7602) exited]
[Thread 0x7fff8cdff700 (LWP 7605) exited]
[Thread 0x7fff8d812700 (LWP 7603) exited]
[New Thread 0x7fff8d812700 (LWP 7606)]
[Thread 0x7fff8d812700 (LWP 7606) exited]
[New Thread 0x7fff8d812700 (LWP 7607)]
[New Thread 0x7fff8cdff700 (LWP 7608)]
[New Thread 0x7fff8e7fc700 (LWP 7610)]
[New Thread 0x7fff67fff700 (LWP 7611)]

Thread 1 "python2" received signal SIGSEGV, Segmentation fault.
0x00007ffff7bc3770 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
(gdb) where
#0  0x00007ffff7bc3770 in pthread_mutex_lock () at /usr/lib/libpthread.so.0
#1  0x00007fffee78171a in XrmQGetResource () at /usr/lib/libX11.so.6
#2  0x00007fffee75dba6 in XGetDefault () at /usr/lib/libX11.so.6
#3  0x00007fffee2a2aef in  () at /usr/lib/libcairo.so.2
#4  0x00007fffee2a5570 in  () at /usr/lib/libcairo.so.2
#5  0x00007fffee27b530 in cairo_surface_get_font_options () at /usr/lib/libcairo.so.2
#6  0x00007fffee22ee12 in  () at /usr/lib/libcairo.so.2
#7  0x00007fffee232144 in  () at /usr/lib/libcairo.so.2
#8  0x00007fffee289fe4 in cairo_show_glyphs () at /usr/lib/libcairo.so.2
#9  0x00007ffff077be4c in  () at /usr/lib/libpangocairo-1.0.so.0
#10 0x00007ffff077c07c in  () at /usr/lib/libpangocairo-1.0.so.0
#11 0x00007ffff12cd2df in pango_renderer_draw_glyph_item () at /usr/lib/libpango-1.0.so.0
#12 0x00007ffff12cf3ea in pango_renderer_draw_layout_line () at /usr/lib/libpango-1.0.so.0
#13 0x00007ffff12cf7d9 in pango_renderer_draw_layout () at /usr/lib/libpango-1.0.so.0
#14 0x00007ffff0778d0b in  () at /usr/lib/libpangocairo-1.0.so.0
#15 0x00007fffeb8fb58b in gtk_render_layout () at /usr/lib/libgtk-3.so.0
#16 0x00007fffeb869b03 in  () at /usr/lib/libgtk-3.so.0
#17 0x00007fffeb7ba8bd in  () at /usr/lib/libgtk-3.so.0
#18 0x00007fffeb7bf5c8 in  () at /usr/lib/libgtk-3.so.0
#19 0x00007fffeb866bd2 in  () at /usr/lib/libgtk-3.so.0
#20 0x00007fffeb74d6c1 in  () at /usr/lib/libgtk-3.so.0
#21 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#22 0x00007fffeb7b55ff in gtk_container_propagate_draw () at /usr/lib/libgtk-3.so.0
#23 0x00007fffeb7b56e3 in  () at /usr/lib/libgtk-3.so.0
#24 0x00007fffeb898a16 in  () at /usr/lib/libgtk-3.so.0
#25 0x00007fffeb7ba8bd in  () at /usr/lib/libgtk-3.so.0
---Type <return> to continue, or q <return> to quit---
#26 0x00007fffeb7bf5c8 in  () at /usr/lib/libgtk-3.so.0
#27 0x00007fffeb898a42 in  () at /usr/lib/libgtk-3.so.0
#28 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#29 0x00007fffeb7b55ff in gtk_container_propagate_draw () at /usr/lib/libgtk-3.so.0
#30 0x00007fffeb7b56e3 in  () at /usr/lib/libgtk-3.so.0
#31 0x00007fffeb895cfd in  () at /usr/lib/libgtk-3.so.0
#32 0x00007fffeb7ba8bd in  () at /usr/lib/libgtk-3.so.0
#33 0x00007fffeb7bf5c8 in  () at /usr/lib/libgtk-3.so.0
#34 0x00007fffeb8967b2 in  () at /usr/lib/libgtk-3.so.0
#35 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#36 0x00007fffeb7b55ff in gtk_container_propagate_draw () at /usr/lib/libgtk-3.so.0
#37 0x00007fffeb7b56e3 in  () at /usr/lib/libgtk-3.so.0
#38 0x00007fffeb766f45 in  () at /usr/lib/libgtk-3.so.0
#39 0x00007fffeb7ba8bd in  () at /usr/lib/libgtk-3.so.0
#40 0x00007fffeb7bf5c8 in  () at /usr/lib/libgtk-3.so.0
#41 0x00007fffeb7698d2 in  () at /usr/lib/libgtk-3.so.0
#42 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#43 0x00007fffeb7b55ff in gtk_container_propagate_draw () at /usr/lib/libgtk-3.so.0
#44 0x00007fffeb7b56e3 in  () at /usr/lib/libgtk-3.so.0
#45 0x00007fffeb766f45 in  () at /usr/lib/libgtk-3.so.0
#46 0x00007fffeb7ba8bd in  () at /usr/lib/libgtk-3.so.0
#47 0x00007fffeb7bf5c8 in  () at /usr/lib/libgtk-3.so.0
#48 0x00007fffeb7698d2 in  () at /usr/lib/libgtk-3.so.0
#49 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#50 0x00007fffeb7b55ff in gtk_container_propagate_draw () at /usr/lib/libgtk-3.so.0
#51 0x00007fffeb7b56e3 in  () at /usr/lib/libgtk-3.so.0
---Type <return> to continue, or q <return> to quit---
#52 0x00007fffeb9ea32f in  () at /usr/lib/libgtk-3.so.0
#53 0x00007fffeb9dc2f0 in  () at /usr/lib/libgtk-3.so.0
#54 0x00007fffeb9e5893 in  () at /usr/lib/libgtk-3.so.0
#55 0x00007fffeb880c0a in gtk_main_do_event () at /usr/lib/libgtk-3.so.0
#56 0x00007ffff09b5526 in  () at /usr/lib/libgdk-3.so.0
#57 0x00007ffff09c5bcb in  () at /usr/lib/libgdk-3.so.0
#58 0x00007ffff09c6e29 in  () at /usr/lib/libgdk-3.so.0
#59 0x00007ffff09c7029 in  () at /usr/lib/libgdk-3.so.0
#60 0x00007ffff5258a4d in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#61 0x00007ffff526be40 in  () at /usr/lib/libgobject-2.0.so.0
#62 0x00007ffff52746f6 in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#63 0x00007ffff5275130 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#64 0x00007ffff09bec5a in  () at /usr/lib/libgdk-3.so.0
#65 0x00007ffff09a9843 in  () at /usr/lib/libgdk-3.so.0
#66 0x00007ffff4f7ecb3 in  () at /usr/lib/libglib-2.0.so.0
#67 0x00007ffff4f7e1d6 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#68 0x00007ffff4f7e5b1 in  () at /usr/lib/libglib-2.0.so.0
#69 0x00007ffff4f7e8e2 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#70 0x00007fffeb87fdcf in gtk_main () at /usr/lib/libgtk-3.so.0
#71 0x00007ffff4d2f1c8 in ffi_call_unix64 () at /usr/lib/libffi.so.6
#72 0x00007ffff4d2ec2a in ffi_call () at /usr/lib/libffi.so.6
#73 0x00007ffff56fcbdb in  () at /usr/lib/python2.7/site-packages/gi/_gi.so
#74 0x00007ffff56fe938 in  () at /usr/lib/python2.7/site-packages/gi/_gi.so
#75 0x00007ffff7454003 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0
#76 0x00007ffff7469e37 in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#77 0x00007ffff75005a6 in PyEval_EvalCodeEx () at /usr/lib/libpython2.7.so.1.0
---Type <return> to continue, or q <return> to quit---
#78 0x00007ffff746cb51 in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#79 0x00007ffff7467550 in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0
#80 0x00007ffff75005a6 in PyEval_EvalCodeEx () at /usr/lib/libpython2.7.so.1.0
#81 0x00007ffff750575a in PyEval_EvalCode () at /usr/lib/libpython2.7.so.1.0
#82 0x00007ffff751d271 in run_mod () at /usr/lib/libpython2.7.so.1.0
#83 0x00007ffff751eb55 in PyRun_FileExFlags () at /usr/lib/libpython2.7.so.1.0
#84 0x00007ffff751ed2a in PyRun_SimpleFileExFlags () at /usr/lib/libpython2.7.so.1.0
#85 0x00007ffff75128d3 in Py_Main () at /usr/lib/libpython2.7.so.1.0
#86 0x00007ffff782106b in __libc_start_main () at /usr/lib/libc.so.6
#87 0x000055555555477a in _start ()
(gdb) 
```

The output of ldd on the mlt libraries is as follows:
```bash
$ for lib in /usr/lib/mlt/* /usr/lib/libmlt*; do ldd $lib | grep -i sdl && echo $lib; done
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fa57e6f8000)
/usr/lib/mlt/libmltavformat.so
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fa2cf6c7000)
/usr/lib/mlt/libmltsdl2.so
	libSDL-1.2.so.0 => /usr/lib/libSDL-1.2.so.0 (0x00007fd8cdc99000)
	libSDL_image-1.2.so.0 => /usr/lib/libSDL_image-1.2.so.0 (0x00007fd8cda7b000)
/usr/lib/mlt/libmltsdl.so
```
The output of 
```
$ pacman -Qi mlt
``` 
says this much about its version and dependencies:
```bash
Name            : mlt
Version         : 6.8.0-2
Description     : An open source multimedia framework
Architecture    : x86_64
URL             : http://www.mltframework.org
Licenses        : GPL
Groups          : None
Provides        : None
Depends On      : libebur128
Optional Deps   : sdl_image: SDL1 plugin [installed]
                  sdl2: SDL2 plugin [installed]
                  libsamplerate: libavresample plugin [installed]
                  sox: SOX (Audio Swiss Army Knife) plugin [installed]
                  ffmpeg: ffmpeg plugin [installed]
                  vid.stab: video stabilize plugin [installed]
                  gtk2: Gtk plugin [installed]
                  qt5-svg: Qt5 plugins [installed]
                  jack: JACK sound output plugin [installed]
                  ladspa: LADSPA plugins [installed]
                  libexif: auto rotate plugin [installed]
                  frei0r-plugins: for additional effects [installed]
                  movit: opengl plugin [installed]
                  opencv: openCV plugin [installed]
Required By     : flowblade  mlt-python-bindings
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 34.15 MiB
Packager        : Antonio Rojas <arojas@archlinux.org>
Build Date      : Fri 11 May 2018 08:43:07 AM CEST
Install Date    : Wed 16 May 2018 11:33:01 AM CEST
Install Reason  : Installed as a dependency for another package
Install Script  : No
Validated By    : Signature
```
and the same for flowblade:
```
$ pacman -Qi flowblade
Name            : flowblade
Version         : 1.16-2
Description     : a multitrack non-linear video editor for Linux
Architecture    : any
URL             : https://github.com/jliljebl/flowblade
Licenses        : GPL3
Groups          : None
Provides        : None
Depends On      : dbus-glib  frei0r-plugins  gmic  gtk3  librsvg  mlt  mlt-python-bindings  movit  pygtk  python2-dbus  python2-gobject  python2-numpy  python2-pillow  sdl_image  sox  swh-plugins
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 22.71 MiB
Packager        : Unknown Packager
Build Date      : Wed 16 May 2018 11:51:08 AM CEST
Install Date    : Wed 16 May 2018 11:51:08 AM CEST
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : None
```

**Comment by rredesigns** [comment]
No. It works fine in Arch (well, I'm on Manjaro, but we have the same issues due to the mixed repos). Difference is, I only installed Flowblade through the package manager to get the dependencies, then I downloaded this repo as a zip and run it from there.

I suggest you do the same. Things from AUR are unnecessary pain.

**Comment by arugifa** [comment]
@dcanelhas How did you reinstall Flowblade? Which package manager did you use?

Installing ``flowblade`` (not ``flowblade-git``) from AUR with ``pacaur`` worked seamlessly for me (``pacaur -S flowblade``). I just add to remove it before (``pacaur -Rns flowblade``).

**Comment by oberon2007** [comment]
Hi there. I'm the maintainer of the AUR package, and I run Manjaro.
For me **flowblade** segfaults, no matter if I use the installed package or try to run run the script directly in the archive.
We currently have **mlt 6.8.0** and **python2 2.7.15**
@rredesigns I have no idea how or why it is working for you...
Can we please reopen this or should we open a new issue @jliljebl ?

**Comment by oberon2007** [comment]
As far as I can tell **flowblade** stopped working for me with the recent update of **mlt** a few days ago.

**Comment by rredesigns** [comment]
Hi Oberon. You are a rockstar for Manjaro users. :D 

Yeah, I installed Flowblade from the official repo just once and long ago. Once I saw this git repo was way ahead, I just started downloading this repo zipped and adding the python command to my menu entry.

I run Flowblade by last time a couple of hours ago and my system is fully up to date.

**Comment by rredesigns** [comment]
I have MLT 6.6.0. And Python 2.7.15. Any chance my system missed the MLT update or something? (I last checked just now and no new updates).

Here's the stdout log:
`[rre@manjaro flowblade-trunk]$ python2 flowblade 
FLOWBLADE MOVIE EDITOR 1.16
---------------------------
Launch script dir: /mnt/Storage/Linux_files/flowblade-1.16/flowblade-trunk
Running from filesystem...
MLT found, version: 6.6.0
numpy version: 1.14.3
OS: Manjaro Linux
Python 2.7.15 (default, May  1 2018, 20:16:04) 
[GCC 7.3.1 20180406]
GTK+ version: 3.22.30
User dir: /home/rre/.flowblade/
Locale: en_US
Found translations at /mnt/Storage/Linux_files/flowblade-1.16/flowblade-trunk/Flowblade/locale/, using those.
Use OS locale language.
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: Flowblade Default
`

**Comment by oberon2007** [comment]
Well, here we go! :stuck_out_tongue_winking_eye: You are on Manjaro stable branch! Looks like **flowblade** will brake for you with the next stable update ... ...
At least we now know that the issue is related to this update :wink:

**Comment by oberon2007** [comment]
Opened a new issue.

**Comment by arugifa** [comment]
I have the same Python, MLT and GTK versions than @rredesigns , I am using ArchLinux, and I can confirm that Flowblade works fine for me too :+1:

**Comment by oberon2007** [comment]
@arugifa then your sytem isn't up-to-date.
Archlinux **mlt** version is [**6.8.0-2**](https://www.archlinux.org/packages/extra/x86_64/mlt/) !
